beacon/weakcoin/interface.go

```diff
@@ -24,3 +26,12 @@ type nonceFetcher interface {
 type allowance interface {
 	MinerAllowance(types.EpochID, []byte) uint32
 }
+
+// weakCoinClock interface exists so that we can pass an object from the beacon
+// package to the weakCoinPackage (as does allowance), this is indicatave of a
+// circular dependency, probably the weak coin should be merged with the beacon
+// package.
+// Issue: https://github.com/spacemeshos/go-spacemesh/issues/4199
+type weakCoinClock interface {
```

Comment:
maybe skip this when generating mocks? it seems you are using your own implementation to mock it

Reply:
Unfortunately it's not possible to use that approach for unexported interfaces, you need to use source mode.

```
Source mode generates mock interfaces from a source file.
It is enabled by using the -source flag. Other flags that
may be useful in this mode are -imports and -aux_files.
Example:
	mockgen -source=foo.go [other options]
```

So it looks like we're back to the 2 options I outlined above, or we could export all these interfaces, but I like that option less than the other two.

Reply:
i think whatever you prefer is ok. i personally would make interfaces public in order for them to work with that source mode, and likely wouldn't put them in interfaces file at all

Reply:
I just moved the weakCoinClock interface out of interface.go.

Given that all these interfaces exist to either allow objects to be injected into the package or to allow mocking (i.e. none of the interfaces represented functionality from within the package), making them public didn't seem like a good option.